kanidmd/src/lib/idm/server.rs

```diff
@@ -1522,11 +1521,30 @@ impl<'a> IdmServerProxyWriteTransaction<'a> {
             au,
             "idm::server::IdmServerProxyWriteTransaction::commit",
             || {
+                if self
+                    .qs_write
+                    .get_changed_uuids()
+                    .contains(&UUID_SYSTEM_CONFIG)
+                {
+                    self.reload_password_badlist(au)?;
+                };
                 self.mfareg_sessions.commit();
                 self.qs_write.commit(au)
             }
         )
     }
+
+    fn reload_password_badlist(&self, au: &mut AuditScope) -> Result<(), OperationError> {
+        match self.qs_write.get_password_badlist(au) {
+            Ok(badlist_entry) => {
+                let mut pw_badlist_cache_write_txn = self.pw_badlist_cache.write();
```

Comment:
I don't think this is quitecorrect. You should take the .write() in the start of the ProxyWrite transaction, so you can just directly call into current value. Then you commit it before qs_write.commit.

Reply:
My thinking was we should get the write lock (.write()) only when the changed_uuid contains UUID_SYSTEM_CONFIG.
But I suppose it doesn't really matter if we are using CowCell and there's only 1 writer?

Reply:
The main reason to "always take it" in the lock form of the transaction is that if a reader was to "mistakenly" be coded to attempt to take the write during a read, there is a possibility of deadlock.

So it's better here given there is only a single writer, to take the write lock at the start of the write, to ensure we really do have access to it, and similar, we take the read lock at the start of a read, so we know we can never accidentaly deadlock during an operation. 

It's more about preventing human/programmer error (we are all likely to make mistakes!) than anything.

And of course, because of the structure of the code, the lock is cheap to take anyway :)

Reply:
Ah good, I see your point!